File: 2025/src/day02/task1.rs
use std::{fs::File, io::Read, time::Instant};

const PATH: &str = "inputs/day02/input.txt";

fn task1() {
    let mut invalid_id_sum = 0;
    let mut text = String::new();
    let _ = File::open(PATH).unwrap().read_to_string(&mut text).unwrap();

    let ranges = text.trim().split(",").map(|range| {
        range
            .split("-")
            .map(|s| s.parse::<i64>().unwrap())
            .collect::<Vec<_>>()
    });

    for range in ranges {
        for id in range[0]..=range[1] {
            let mut width = 0;
            while 10_i64.pow(width) <= id {
                width += 1;
            }

            if width % 2 == 1 {
                continue;
            }

            let cutoff = 10_i64.pow(width / 2);
            let left = id / cutoff;
            let right = id % cutoff;

            if left == right {
                invalid_id_sum += id;
            }
        }
    }
    println!("{invalid_id_sum}")
}

fn main() {
    let timer = Instant::now();
    task1();
    println!("Done in {:?}", timer.elapsed());
}

File: 2017/src/day10/task1.ts
import path from "node:path";

const PATH = path.join("inputs", "day10", "input.txt");
const LENGTH = 256;

async function task1() {
   const input = (await Bun.file(PATH).text()).trim().split(",").map(Number);
   const list = Array.from({ length: LENGTH }).map((_, i) => i);

   let current = 0;
   let skip = 0;

   for (const offset of input) {
      if (offset > LENGTH) {
         throw new Error("Invalid input");
      }
      const endIndex = (current + offset - 1) % LENGTH;

      for (let i = 0; i < Math.floor(offset / 2); i++) {
         const left = (current + i) % LENGTH;
         let right = endIndex - i;
         if (right < 0) {
            right += LENGTH;
         }

         list[left]! ^= list[right]!;
         list[right]! ^= list[left]!;
         list[left]! ^= list[right]!;
      }

      current = (current + offset + skip) % LENGTH;
      skip += 1;
   }

   console.log(list[0]! * list[1]!);
}

console.time("Task Done");
await task1();
console.timeEnd("Task Done");
